l1-contracts/contracts/bridge/L1SharedBridge.sol

```diff
@@ -946,7 +953,8 @@ contract L1SharedBridge is IL1SharedBridge, ReentrancyGuard, Ownable2StepUpgrade
         uint16 _l2TxNumberInBatch,
         bytes32[] calldata _merkleProof
     ) external override onlyLegacyBridge {
-        bytes memory transferData = abi.encode(_amount, _depositSender);
+        // For legacy deposits, the l2 receiver is not required to check tx data hash
+        bytes memory transferData = abi.encode(_amount, address(0));
```

Comment:
How will this be correct though? setting an incorrect address(0) there will ruin the encoding

Reply:
@kelemeno why? This is a legacy function. For them, the l2 receiver is never used inside the encoding

Reply:
Ah ok